Cargo.toml

```
@@ -43,14 +43,26 @@ url = { version = "2.2.2", features = ["serde"] }
 impl-serde = "0.3.2"
 regex = "1.5.4"
 
-# dependencies for optional extrinsics feature
-async-std = { version = "1.10.0", optional = true }
-sp-core = { version = "2.0.1", optional = true }
-subxt = { version = "0.14.0", package = "substrate-subxt", optional = true }
-futures = { version = "0.3.21", optional = true }
-hex = { version = "0.4.3", optional = true }
-
-funty = "2.0.0"
+# dependencies for extrinsics (deploying and calling a contract)
+async-std = { version = "1.10.0", features = ["attributes", "tokio1"] }
+ink_metadata = { version = "3.0.0-rc8", features = ["derive"] }
+ink_env = "3.0.0-rc8"
+scale-info = { version = "1.0.0", default-features = false, features = ["derive"] }
+sp-arithmetic = "4.0.0"
+sp-core = "5.0.0"
+sp-keyring = "4.0.0"
```

Those pulled in a lot of duplicated dependencies and are actually unused in the code.

```suggestion
sp-core = "5.0.0"
```